### PR TITLE
Add Nbody sample to the oneAPI samples repo

### PR DESCRIPTION
**Pull request by mpanoop**
# Description

All the feedbacks and review comments received from reviewers are incorporated here.

Fixes # (issue) 

## Type of change

Please delete options that are not relevant. Add a 'X' to the one that is applicable. 

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Sample Migration (Moving sample from old repository after completing checklist established)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Command Line
- [ ] oneapi-cli
- [X] Visual Studio
- [ ] Eclipse IDE
- [ ] VSCode

# Checklist for Moving samples:
Links and Details can be found in the samples WG Teams Files. 

- [ ] Review sample design with domain reviewers https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts 
- [ ] Implement coding guidelines and ensure code quality.
- [ ] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [ ] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com
- [ ] Adhere to readme template 
- [ ] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [ ] Enforce format via clang-format config file
- [ ] Review DPC++ code with Paul Peterseon. (GitHub User: pmpeter1)
- [ ] Review readme with Tom Lenth or Joe Oster. (GitHub User: JoeOster)
- [ ] Tested using Dev Cloud when applicable
- [ ] Implement fixes for ONSAM Jiras
- [ ] If you have new dependencies/binaries, inform Samples Project Manager Swapna R Dontharaju (@srdontha) or @JoeOster

### CODE REVIEW

File: DirectProgramming/DPC++/N-BodyMethods/Nbody/src/GSimulation.cpp [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+// =============================================================
+
+// =============================================================
+// Copyright (c) 2019 Fabio Baruffa
+// Source: https://github.com/fbaru-dev/particle-sim
+// MIT License
+// =============================================================
+
+#include "GSimulation.hpp"
+// dpc_common.hpp can be found in the dev-utilities include folder.
```
Comment (JoeOster) [inline DirectProgramming/DPC++/N-BodyMethods/Nbody/src/GSimulation.cpp:14]: The default style is there should be a blank line between the include and the comment

File: DirectProgramming/DPC++/N-BodyMethods/Nbody/src/GSimulation.cpp [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+// e.g., $ONEAPI_ROOT/dev-utilities/latest/include/dpc_common.hpp
+#include "dpc_common.hpp"
+using namespace sycl;
```
Comment (JoeOster) [inline DirectProgramming/DPC++/N-BodyMethods/Nbody/src/GSimulation.cpp:17]: A blank line here as well

File: DirectProgramming/DPC++/N-BodyMethods/Nbody/src/GSimulation.cpp [hunk 3]
```diff
@@ -0,0 +1,230 @@
+//==============================================================
```
Comment (JoeOster) [inline DirectProgramming/DPC++/N-BodyMethods/Nbody/src/GSimulation.cpp:1]: Was this put through clang format?

Comment (mpanoop) [reply DirectProgramming/DPC++/N-BodyMethods/Nbody/src/GSimulation.cpp:1]: Yes, This was out through clang format with Google style.

File: DirectProgramming/DPC++/N-BodyMethods/Nbody/src/GSimulation.cpp (hunk 1 above)
Comment (mpanoop) [reply DirectProgramming/DPC++/N-BodyMethods/Nbody/src/GSimulation.cpp:9]: This change is done and the commit is pushed to this PR

File: DirectProgramming/DPC++/N-BodyMethods/Nbody/src/GSimulation.cpp (hunk 2 above)
Comment (mpanoop) [reply DirectProgramming/DPC++/N-BodyMethods/Nbody/src/GSimulation.cpp:13]: This change is already done and pushed in.

**Pull request merged**